frontend/src/components/explorer/mapViewer/ThreeDviewer.vue

```diff
@@ -306,7 +310,10 @@ export default {
             group: n.g,
           }));
 
-        if (this.searchedNodesOnMap.length > 0) {
+        if (centerId) {
+          const centerNode = this.searchedNodesOnMap.filter(n => n.id === centerId)[0];
```

Comment:
I do not have the context here, but I generally like the idea of [assertions](https://www.informit.com/articles/article.aspx?p=2982114&seqNum=3). If it is the case that we always expect exactly one of the nodes to have `centerId` as `id` then I think something like this could be a good thing: 

```
const matches = this.searchedNodesOnMap.filter(n => n.id === centerId);
if (matches.length !== 1) {
   throw new Error(`Expected exactly one node with id ${centerId}, but found ${matches}`);    
}
await this.centerElement(matches[0]);
```

Reply:
To my knowledge, the 3D maps do not contain any duplicates at the moment. However, it is envisioned that they will, as it would enable better layouts. So, in the end, both 2D maps and 3D maps will have duplicates.

Reply:
I like that this is less nested and looks quite clean @MalinAhlberg! I think technically it becomes "center the first search result, then center it on node matching centerId if possible", which can result in two `centerElement` calls. I think the below would be more like what we did before but in your new code style.

```javascript
        // center the selected node if there is such, else the first search result
        let matches = this.searchedNodesOnMap;
        if (centerId) {
          matches = this.searchedNodesOnMap.filter(n => n.id === centerId);
        }
        if (matches.length > 0) {
          await this.centerElement(matches[0]);
        }
```

Which can be further reduced to the following if you prefer.

```javascript
        // center the selected node if there is such, else the first search result
        const matches = centerId ? this.searchedNodesOnMap.filter(n => n.id === centerId) : this.searchedNodesOnMap;

        if (matches.length > 0) {
          await this.centerElement(matches[0]);
        }
```

Reply:
Oops, yes, you first version is what I intended to write, must have copy-pasted the wrong line...
Anyway, your second version is short and clean, I'll go for that. Thanks!

Reply:
> Oops, yes, you first version is what I intended to write, must have copy-pasted the wrong line...

Makes sense! I suspected that was probably the case. Really nice suggestion from you again as it handles the `matches` length for both cases now.